dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/MavenFacade.scala

```diff
@@ -0,0 +1,276 @@
+        StringUtils.isEmpty(dep.getArtifactId) ||
+        StringUtils.isEmpty(dep.getVersion)
+      )) {
+        collect.addDependency(RepositoryUtils.toDependency(dep, stereotypes))
+      }
+    }
+
+    // Add additional dependencies
+    additionalDependencies.foreach(collect.addDependency)
+
+    val depMngt = project.getDependencyManagement
+    if (depMngt != null) {
+      depMngt.getDependencies.asScala.foreach { dep =>
+        collect.addManagedDependency(RepositoryUtils.toDependency(dep, stereotypes))
+      }
+    }
+
+    val depResult = resolveDependencies(collect)
+
+    // The code below comes from org.apache.maven.project.DefaultProjectBuilder
+    val artifacts = new util.LinkedHashSet[org.apache.maven.artifact.Artifact]
+    if (depResult.getRoot != null) {
+      RepositoryUtils.toArtifacts(artifacts, depResult.getRoot.getChildren, Collections.singletonList(project.getArtifact.getId), null)
+      val lrm = session.getRepositorySession.getLocalRepositoryManager
+      artifacts.asScala.foreach { artifact =>
+        if (!artifact.isResolved) {
+          val path = lrm.getPathForLocalArtifact(RepositoryUtils.toArtifact(artifact))
+          artifact.setFile(new File(lrm.getRepository.getBasedir, path))
+        }
+      }
+    }
+    project.setResolvedArtifacts(artifacts)
+    project.setArtifacts(artifacts)
+
+    toDependencies(depResult)
+  }
+
+  private def resolveDependencies(collect: CollectRequest): DependencyResult = {
+    val depRequest = new DependencyRequest(collect, null)
+
+    // Replace the workspace reader with one that will resolve projects that haven't been compiled yet
+    val repositorySession = new DefaultRepositorySystemSession(session.getRepositorySession)
+    repositorySession.setWorkspaceReader(new UnbuiltWorkspaceReader(repositorySession.getWorkspaceReader, session))
+
+    val collectResult = repoSystem.collectDependencies(repositorySession, collect)
+
+    val node = collectResult.getRoot
+    depRequest.setRoot(node)
+
+    repoSystem.resolveDependencies(repositorySession, depRequest)
+  }
+
+  private def toDependencies(depResult: DependencyResult): Seq[Dependency] = {
+    depResult.getArtifactResults.asScala.map(_.getRequest.getDependencyNode.getDependency)
+  }
+
+  def locateServices: Seq[MavenProject] = {
+    session.getAllProjects.asScala.filter(isService)
+  }
+
+  private def isService(project: MavenProject): Boolean = {
+    // If the value is set, return it
+    isLagomOrPlayService(project).getOrElse {
+
+      // Otherwise try and run lagom:configure
+      if (executeMavenPluginGoal(project, "configure")) {
+
+        // Now try and get the value
+        isLagomOrPlayService(project).getOrElse {
+
+          // The value should have been set by lagom:configure, fail
+          sys.error(s"${LagomKeys.LagomService} not set on project ${project.getArtifactId} after running configure!")
+        }
+      } else {
+        // Lagom plugin not configured, return false
```

Comment:
Do we want to add a log.debug statement here for making it easier to understand why their defined project isn't picked up as a service?